src/blocksparsearray/blockdiagonalarray.jl

```diff
@@ -0,0 +1,30 @@
+# type alias for block-diagonal
+using LinearAlgebra: Diagonal
+
+const BlockDiagonal{T,A,Axes,V<:AbstractVector{A}} = BlockSparseMatrix{
```

Comment:
What about calling it `BlockSparseDiagonal` to disambiguate from `BlockArrays.BlockDiagonal`?

Reply:
I think the thing that gives me pause about this design is that ideally this would also handle other kinds of diagonal types, like `DiagonalArrays.DiagonalMatrix` and `FillArrays.RectDiagonal`, but it shouldn't be too hard to generalize the functionality to work generically across those (i.e. by expanding the type signatures and replacing some parts that are specific to the `LinearAlgebra.Diagonal` API with more generic APIs from `DiagonalArrays.jl`).

Reply:
I do agree that we should work on supporting that, but probably not in this PR, to keep things comprehensible and moving :)

Reply:
> I do agree that we should work on supporting that, but probably not in this PR, to keep things comprehensible and moving :)

Agreed, we can worry about that in future PRs, I just find it helpful for reviewing to think ahead a little bit to try to anticipate if any of the current designs won't generalize well to cases we want to cover.